sklearn/neighbors/base.py

```diff
@@ -334,11 +350,39 @@ class from an array representing our data set and ask who's
                     "or set algorithm='brute'" % self._fit_method)
             result = self._tree.query(X, n_neighbors,
                                       return_distance=return_distance)
-            return result
         else:
             raise ValueError("internal: _fit_method not recognized")
 
-    def kneighbors_graph(self, X, n_neighbors=None,
+        if query_is_train:
```

Comment:
Put the shorter case first:
```
if not query_is_train:
    return result
else:
    ...
```
(I'm also okay with having no else but I've seen that disperefered in the project)